### PR TITLE
16 - Install grommet + theme

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number
#16  

**Mockup:** [refinebio Styles Components](https://www.figma.com/file/FCfyrM8yzJqa1QE60Rlksk/refinebio-styles-components)

## Purpose/Implementation Notes

Install grommet and set up theming.

Please visit [Vercel Preview](https://refinebio-web-git-nozomione-16-install-grommet-and-605b14-ccdl.vercel.app) to review the UI(@dvenprasad). 

Updates include:

- [x] Install and integrate Grommet in the project
- [x] Create the `src/themes` folder for customization
- [x] Set up the initial colors and fonts
- [x] Define building-block themes/components

<details>
<summary>(Files include)</summary>   
 
**Themes(in `.src/themes/`):** 
- `global`, `colors`
- (font styles) `Heading`, `Paragraph`, `Text`, `Anchor`(for links)
- (buttons) `Button`, `Tab`/`Tabs`(for tabbed buttons),
- (form elements) `TextInput`, `Select`, `RadioButton`, `CheckBox`
- (navigation) `Menu`
    
    
**Components**
**(in `./components/shared/`):** 
- (form elements) Input (a wrapper component for `InputText`), Tag (for a filter tag)
- Button (a wrapper component for Grommet `Button`)
- Tabs (a wrapper component for Grommet `Tabs`)
- (badge) IconBadge, Pill
- (utility) InlineMessage

 **(in `./components/header/`):** 
- Logo (Refinebio site logo)
- NavDropDown(will be re-defined when creating the composite component for `Header`)
- NavLInk(will be re-defined when creating the composite component for `Header`)

**others:**
- `themes/variable.js` which includes constant variables for form elements
- SVG image: **12**
- Auto-generated artifact: **1** (`yarn.lock`)
- Temporary edited: **1** (`pages/index.js`) 
</details>
   

**NOTE:** We'll create other variables/components required in future implementation.

Thank you, David!

## Checklist
- [x] Lint and unit tests pass locally with my changes

### CODE REVIEW

**Comment by nozomione** [comment]
@dvenprasad Please review the UI update at [Vercel Preview](https://refinebio-web-git-nozomione-16-install-grommet-and-605b14-ccdl.vercel.app/), thank you!

**Comment by dvenprasad** [comment]
@nozomione Most of the changes look good! There were two things I noticed:

- Under inline messages, the icon and text do not seem horizontally centered for success and info variations

Screen Shot 2022-09-26 at 11 15 35 AM

- For the navigation links, are there `selected` states which are just toggled on here?

**Comment by nozomione** [comment]
@dvenprasad
 
>   Under inline messages, the icon and text do not seem horizontally centered for success and info variations

Adjusted ✅ 

>   For the navigation links, are there `selected` states which are just toggled on here?

Yes, when the global navigation is utilized with pages, it would be `selected`.

Regarding the **number badge**, I left you a message via skack. Please ping me when you get a chance, thank you!

**Comment by nozomione** [comment]
@davidsmejia This PR will be ready for your review once the UI review is approved(added @dvenprasad to one of the reviewers). Thank you!

File: docker-compose.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ services:
       dockerfile: Dockerfile
     command: >
       bash -c 'cd /app &&
-             yarn && 
+             yarn install --silent &&
```
Comment (davidsmejia) [inline docker-compose.yml:13]: not so sure we want to do this here

Comment (nozomione) [reply docker-compose.yml]: Completed - removed the `--silent` flag and now simply install the dependencies from a lock file. Thank you!

**Comment by nozomione** [comment]
Temporarily converted back to Draft until the following list is complete;
- Remove `--frozen-lockfile` flag from `yarn` command in docker-compose
- Eliminate the use of the CSS class and instead use the props to extend the theme variants

File: src/components/shared/Pill.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <Box
+      align="center"
+      background={status ? `${status}-bg` : background}
+      direction="row"
+      flex
+      pad={{ horizontal: 'xsmall', vertical: 'xxsmall' }}
+      round="30px"
+      width="max-content"
+    >
+      {!labelOnly && (
```
Comment (davidsmejia) [inline src/components/shared/Pill.js:22]: I would change the name of this to be `dot` and when true, have it show the dot.

File: src/components/shared/InlineMessage.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        <Box margin={{ right: '8px' }} height="inherit">
+          <Text
+            color={status === 'error' ? errorIconColor : status}
+            height="24px"
+          >
+            <SVGIcon role="presentation" aria-hidden="true" focusable="false" />
+          </Text>
+        </Box>
+      )}
+      <P status={status} size="small">
```
[the remark this answers is not in the document]
Comment (nozomione) [reply src/components/shared/InlineMessage.js]: Completed, thank you!

File: src/components/shared/Pill.js (hunk 2 above)
Comment (nozomione) [reply src/components/shared/Pill.js]: Completed, thank you!

**Comment by nozomione** [comment]
@davidsmejia, based on our chat, merging to dev and closing this PR, thank you!

**Pull request merged**